### DISCUSSION TITLE
SharePoint PnP Search v4 - Not able to filter by user's Office location/city (PhysicalDeliveryOfficeName from AD)

### DISCUSSION THREAD

**Discussion by jussihi** [microsoft-search/pnp-modern-search#3437] [Q&A]
Hello!

I created an employee directory using the PnP Modern Search v4. It works very well, but I would like to filter the results by:

- Department
- Job title
- Physical office location

I can do the first two, but the physical office location seems to be impossible to implement out of the box. I found some answers regarding my question, but all of them required to create some custom attributes/properties to the user profiles in SharePoint. However, with information from Microsoft ([Properties that are synced into SharePoint user profiles](https://learn.microsoft.com/fi-fi/sharepoint/user-profile-sync#properties-that-are-synced-into-sharepoint-user-profiles)) it seems like I could add the location to my local AD attribute "PhysicalDeliveryOfficeName" and then have it in the "Office" property in my SharePoint user instance. However, there is no such property to be selected in the "PnP search results" webpart. Also, there is no such property to be used in the "PnP Search Filters" webpart.

When I check the "Office" property in the user profile properties management in SharePoint admin site, I can see that 35 profiles (out of some 600) have this property set. However, checking the "PhysicalDeliveryOfficeName" from the local AD I think that only 29 profiles should have this property set. How should I even debug this (i.e. see which users have this property set), as I was expecting only 29 users having this property set?

The "Office Location" property (Location) (SPS-Location) is set in 0 profiles. Why? Would it be possible to link the AD "PhysicalDeliveryOfficeName" to this property?

And of course, what would be the easiest way to filter users in a search by their city (set in local AD)?

**Comment by jussihi** [comment]
Edit: I tried to use [Sync-PnPSharePointUserProfilesFromAzureActiveDirectory](https://pnp.github.io/powershell/cmdlets/Sync-PnPSharePointUserProfilesFromAzureActiveDirectory.html) to get this thing working.

The command I tried: `Sync-PnPSharePointUserProfilesFromAzureActiveDirectory -UserProfilePropertyMapping @{"SPS-Location"="city"}` 

However, it failed to `Sync-PnPSharePointUserProfilesFromAzureActiveDirectory: Property names [SPS-Location] are taxonomic properties.` (Or something like that, the error is in Finnish at my end).

**Comment by kasperbolarsen** [comment]
Based on the data in my Dev Tenant it seems like the Office user property is for an Office number, and Office Location (connected to the Term Store) is intended for geographical info like city or region:

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/d12b0d2b-e211-46db-8168-8b2cbd90ee57)

And once the data is indexed the Managed Property BaseOfficeLocation should cover your needs.

**Reply by kasperbolarsen** [reply]
like this :-)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/baffaf63-b667-4cd2-90cb-18f9ba915e39)

**Reply by jussihi** [reply]
Hi @kasperbolarsen and thank you very much for your quick reply!

Unfortunately, I don't have "BaseOfficeLocation" (or "Nationality") in selectable properties in "PnP - Search Results" or "PnP - Search Filters" by default. How can I get them visible and usable?

Also, what is "Term Store"?

**Reply by jussihi** [reply]
To add, my SPS-Location's settings look identical to yours in your first screenshot.

**Reply by kasperbolarsen** [reply]
Are you sure you don't have a BaseOfficeLocation managed prop? I have just checked another tenant and it has that property.


 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/cea02f27-217c-4292-bd76-5025eb57e6e9)



The term store is the masterdata repository subsystem in SharePoint (https://learn.microsoft.com/en-us/sharepoint/create-and-manage-terms)

**Reply by jussihi** [reply]
@kasperbolarsen Yes, I do have that one, but when I use "PnP Search Results" web part with SharePoint Search and "LocalPeopleResults" as the search type, I cannot find the BaseOfficeLocation from the given properties (BaseType is the closest one alphabetically):


![image](https://github.com/microsoft-search/pnp-modern-search/assets/25184163/64ea3eb8-2e33-4866-ac74-a50594172de6)

**Reply by kasperbolarsen** [reply]
Let's start with verifying if there is any data in the BaseOfficeLocation property. 
I suggest using either SP Editor or SharePoint Search Query tool as described here: https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/ 
If the BaseOfficeLocation property does not contain any data you will have to update a few (minimum 5 IFAIK) User Profiles with a value in the Office Location property
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/03deb2d5-f384-490c-a6e8-a9cd07fff733)

and either wait for up to 24 hours and then check again or update the user profiles and run the Reindex script from https://pnp.github.io/script-samples/?query=reindex

**Reply by jussihi** [reply]
@kasperbolarsen Thank you! I will add some values and wait until tomorrow :)

 Is this the reason why my ` Sync-PnPSharePointUserProfilesFromAzureActiveDirectory` script fails when I try to use the `BaseOfficeLocation` as the destination property?

```powershell
Sync-PnPSharePointUserProfilesFromAzureActiveDirectory -UserProfilePropertyMapping @{"BaseOfficeLocation"="city"}
Sync-PnPSharePointUserProfilesFromAzureActiveDirectory: Property names [BaseOfficeLocation] cannot be found from user profiles.
```

**Reply by jussihi** [reply]
@kasperbolarsen thank you, I was able to find that no profile had their BaseOfficeLocation set. Is there any way to set this value/map this value to a value that is already present in AD? Also, is there a way to set someone else's BaseOfficeLocation? Now I can only set my own via https://<mytenant>-my.sharepoint.com//_layouts/15/editprofile.aspx

This SharePoint thing is kind of hard to grasp without knowing what is happening under the hood.... 😃

**Comment by jussihi** [comment]
@kasperbolarsen , I'm sorry but I must give up trying to make this thing working...
